### DISCUSSION TITLE
It says no script found when there clearly is a script. What am I doing wrong?

### DISCUSSION THREAD

**Discussion by Etceterae** [autokey/autokey#575] [Q&A]
I'm trying to run a test script, but when I type `autokey-run -s test.py`, this happens:

`Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/dbus/service.py", line 711, in _message_cb
    retval = candidate_method(self, *args, **keywords)
  File "/usr/lib/python3/dist-packages/autokey/common.py", line 78, in run_script
    self.app.service.run_script(name)
  File "/usr/lib/python3/dist-packages/autokey/service.py", line 228, in run_script
    script = self.__findItem(name, model.Script, "script")
  File "/usr/lib/python3/dist-packages/autokey/service.py", line 236, in __findItem
    raise Exception("No %s found with name '%s'" % (typeDescription, name))
Exception: No script found with name 'test.py'`

There is a script called test.py, and it is in the same folder that the terminal is in. What am I doing wrong?

**Comment by josephj11** [comment]
I don't use this feature, so here are some general comments.

How does AutoKey find test.py? (what path?) What happens if you give it an absolute path?

Does .test.json exist in the same location?

**Reply by Etceterae** [reply]
I'm in the ~/Documents path, the same path as the file. I tried ~/Documents/test.py and /home/(my username)/Documents/test.py but it still doesn't work.

No, .test.json doesn't exist.

**Reply by josephj11** [reply]
I don't use autokey-run. One of the devs will have to tell you how it works. I'm not sure that what follows is relevant to your situation.

So, where did test.py come from? if you wrote it externally to AutoKey, then it will have no metadata file telling AutoKey what to do with it. I don't know the source code, but I suspect that AutoKey doesn't know about your PATH and it's not running in your current directory.

Create a new script inside AutoKey. Open test.py in an editor, etc. and copy and paste it into the script contents panel in AutoKey. Then, set all the parameters for the script in the dialog below and select Save from the main menu. This will let AutoKey know about your script and build the json metadata file for it.

**Reply by BlueDrink9** [reply]
From memory, autokey-run only works with phrases you have already set up. However, if you use the 0.96 (beta) versions, autokey-run can accept an absolute path as an argument and it will run that script through autokey.
